fec.config.js

```diff
@@ -1,27 +1,56 @@
-const { resolve } = require('path');
+const path = require("path");
+const { dependencies, insights } = require("./package.json");
+
+const moduleName = insights.appname.replace(/-(\w)/g, (_, match) =>
+    match.toUpperCase()
+);
+const srcDir = path.resolve(__dirname, "./src");
 
 module.exports = {
-    appUrl: '/insights/patch',
-    debug: true,
-    useProxy: process.env.PROXY === 'true',
-    proxyVerbose: true,
-    plugins: [],
-    ...(process.env.HOT ? { hotReload: process.env.HOT === 'true' } : { hotReload: true }),
+    appName: moduleName,
+    appUrl: "/insights/patch",
+    useProxy: true,
+    ...(process.env.HOT
+        ? { hotReload: process.env.HOT === "true" }
+        : { hotReload: true }),
     ...(process.env.port ? { port: parseInt(process.env.port) } : {}),
     moduleFederation: {
+        moduleName,
+        exposes: {
+            "./RootApp": path.resolve(__dirname, "./src/AppEntry"),
+            "./SystemDetail": path.resolve(__dirname, "./src/index.js"),
+        },
         shared: [
             {
-                'react-router-dom': {
+                "react-router-dom": {
                     singleton: true,
                     import: false,
-                    version: '^6.8.1',
-                    requiredVersion: '>=6.0.0 <7.0.0'
-                }
-            }
+                    version: dependencies["react-router-dom"],
+                    requiredVersion: ">=6.0.0 <7.0.0",
+                },
+            },
+            {
+                "@unleash/proxy-client-react": {
+                    version: dependencies["@unleash/proxy-client-react"],
+                    singleton: true,
+                },
+            },
         ],
-        exposes: {
-            './RootApp': resolve(__dirname, './src/AppEntry'),
-            './SystemDetail': resolve(__dirname, './src/index.js')
-        }
-    }
+    },
+    /**
+     * Add additional webpack plugins
+     */
+    resolve: {
```

Comment:
Why is this needed?

Reply:
Maybe this is handled by chroming, if so it can be removed. 
My understanding is it tells the parent app where our node_modules folder is. 
It was included in the template code I used for my application.
@Hyperkid123?

Reply:
Unless your project resolves modules from a different directory, this is not needed.